### PR TITLE
[EP-637] Customize Card - Handle navigation

### PR DESCRIPTION
**Pull request by mpetrenco**

### CODE REVIEW

File: Source/Modules/PaymentMethods/Router/JPPaymentMethodsRouter.h [hunk 1]
```diff
@@ -35,6 +35,11 @@
  */
 - (void)navigateToTransactionModule;
 
+/**
+ * A method that opens the Card Customization view for customizing the card
```
Comment (gcojocaru) [inline Source/Modules/PaymentMethods/Router/JPPaymentMethodsRouter.h]: Can you add documentation for index param?

File: Source/Modules/PaymentMethods/Presenter/JPPaymentMethodsPresenter.h [hunk 2]
```diff
@@ -37,8 +37,11 @@
 /**
  * A method called once a card was selected from the list. This method updates the view model for that specific card
  * and changes its isSelected state.
+ *
+ * @param index - the index of the card that has been selected
+ * @param isEditing - a boolean property that specifies if the card should be edited
  */
-- (void)didSelectCardAtIndex:(NSInteger)index;
+- (void)didSelectCardAtIndex:(NSInteger)index forEditing:(BOOL)isEditing;
```
Comment (stpopa) [inline Source/Modules/PaymentMethods/Presenter/JPPaymentMethodsPresenter.h]: `- (void)didSelectCardAtIndex:(NSInteger)index inEditingMode:(BOOL)isEditing` ?

File: Source/Modules/PaymentMethods/View/Subviews/Cells/CardCell/JPPaymentMethodsCardCell.m [hunk 3]
```diff
@@ -98,36 +98,53 @@ - (void)configureWithViewModel:(JPPaymentMethodsModel *)viewModel {
...
+    [self setupStackView];
+    [self setupDisclosureIndicator];
+}
+
+- (void)setupIconView {
+    [self.iconContainerView addSubview:self.iconImageView];
+    [self.iconImageView pinToView:self.iconContainerView withPadding:8.0f];
+
+    [NSLayoutConstraint activateConstraints:@[
+        [self.iconContainerView.heightAnchor constraintEqualToConstant:36.0f],
```
Comment (stpopa) [inline Source/Modules/PaymentMethods/View/Subviews/Cells/CardCell/JPPaymentMethodsCardCell.m]: please extract constants

File: Source/Modules/PaymentMethods/View/Subviews/Cells/CardCell/JPPaymentMethodsCardCell.m [hunk 4]
```diff
@@ -98,36 +98,53 @@ - (void)configureWithViewModel:(JPPaymentMethodsModel *)viewModel {
...
-        [self.iconContainerView.widthAnchor constraintEqualToConstant:52.0f],
-    ]];
+    [NSLayoutConstraint activateConstraints:constraints withPriority:999];
+}
+
+- (void)setupDisclosureIndicator {
+    UIImage *disclosureIcon = [UIImage imageWithIconName:@"disclosure-icon"];
+    UIImageView *disclosureImageView = [[UIImageView alloc] initWithImage:disclosureIcon];
+    disclosureImageView.contentMode = UIViewContentModeScaleAspectFit;
+    disclosureImageView.frame = CGRectMake(0, 0, 24, 24);
```
Comment (stpopa) [inline Source/Modules/PaymentMethods/View/Subviews/Cells/CardCell/JPPaymentMethodsCardCell.m]: please extract constants

File: Source/Models/JudoKitObjC.h [hunk 5]
```diff
@@ -39,6 +39,11 @@ FOUNDATION_EXPORT const unsigned char JudoKitObjCVersionString[];
 #import <JudoKitObjC/JPBillingCountry.h>
 #import <JudoKitObjC/JPBrowser.h>
 #import <JudoKitObjC/JPCard.h>
+#import <JudoKitObjC/JPCardCustomizationBuilder.h>
```
Comment (stpopa) [inline Source/Models/JudoKitObjC.h]: `JPCardCustomization*` it's our internal stuff, we don't need to expose this

File: Source/Modules/PaymentMethods/View/Subviews/Cells/CardCell/JPPaymentMethodsCardCell.m [hunk 6]
```diff
@@ -109,36 +116,53 @@ - (void)configureWithViewModel:(JPPaymentMethodsModel *)viewModel {
...
 - (void)setupViews {
     self.backgroundColor = UIColor.clearColor;
+    [self setupIconView];
+    [self setupStackView];
+    [self setupDisclosureIndicator];
+}
+
+- (void)setupIconView {
+    [self.iconContainerView addSubview:self.iconImageView];
+    [self.iconImageView pinToView:self.iconContainerView withPadding:8.0f];
```
Comment (stpopa) [inline Source/Modules/PaymentMethods/View/Subviews/Cells/CardCell/JPPaymentMethodsCardCell.m]: please extract `8.0f` and `3.0` from here and below

File: Source/Modules/PaymentMethods/Presenter/JPPaymentMethodsPresenter.h [hunk 7]
```diff
@@ -37,8 +37,12 @@
 /**
  * A method called once a card was selected from the list. This method updates the view model for that specific card
  * and changes its isSelected state.
+ *
+ * @param index - the index of the card that has been selected
+ * @param isEditing - a boolean property that specifies if the card should be edited
  */
-- (void)didSelectCardAtIndex:(NSInteger)index;
+- (void)didSelectCardAtIndex:(NSInteger)index
```
Comment (gcojocaru) [inline Source/Modules/PaymentMethods/Presenter/JPPaymentMethodsPresenter.h]: Can we change it to NSUInterger to keep it consistent ?

File: Source/Modules/PaymentMethods/Presenter/JPPaymentMethodsPresenter.m [hunk 8]
```diff
@@ -64,7 +64,14 @@ - (void)viewModelNeedsUpdateWithAnimationType:(AnimationType)animationType {
     [self.view configureWithViewModel:self.viewModel];
 }
 
-- (void)didSelectCardAtIndex:(NSInteger)index {
+- (void)didSelectCardAtIndex:(NSInteger)index
```
Comment (gcojocaru) [inline Source/Modules/PaymentMethods/Presenter/JPPaymentMethodsPresenter.m]: Can we change it to NSUInterger to keep it consistent ?

**Pull request merged**